packages/task/src/node/task.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from 'inversify';
+import { TaskManager } from './task-manager';
+import { Process } from '@theia/process/lib/node/process';
+
+@injectable()
+export class Task {
+    protected taskId: number;
+    protected taskProcess: Process;
+    // linter: string | undefined;
+
+    constructor(
+        @inject(TaskManager) protected readonly taskManager: TaskManager,
+        process: Process,
+        linter: string | undefined
+    ) {
+        this.taskId = this.taskManager.register(this);
+        this.taskProcess = process;
+
+        this.taskProcess.onExit(event => {
```

Comment:
You also need to add ProcessManager.delete(this.taskProcess)
And cleanup the onExit listener.

Maybe this could be doner better in general we need to think about this.

Reply:
done both